### PR TITLE
ebs br: add check wal warmup option

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how does it work?

This PR provide restore capability to check wal corruption only during restore by specifying the warmup strategy of volumerestore to `check-wal-only`

cc #5582

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/backup/restore/restore_manager.go [hunk 1]
```diff
@@ -399,7 +401,8 @@ func (rm *restoreManager) checkTiFlashAndTiKVReplicasFromBackupMeta(r *v1alpha1.
 		metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas == 0 {
 		klog.Errorf("backup source tc has no tikv nodes")
 		return fmt.Errorf("backup source tc has no tivk nodes")
-	} else if tc.Spec.TiKV.Replicas != metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas {
+	} else if tc.Spec.TiKV.Replicas != metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas ||
```
Comment (BornChanger) [inline pkg/backup/restore/restore_manager.go:404]: useless change. remove this line.

File: pkg/backup/restore/restore_manager.go [hunk 2]
```diff
@@ -399,7 +401,8 @@ func (rm *restoreManager) checkTiFlashAndTiKVReplicasFromBackupMeta(r *v1alpha1.
...
+		(r.Spec.TolerateSingleTiKVOutage && metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas-1 == tc.Spec.TiKV.Replicas) {
```
Comment (YuJuncen) [inline pkg/backup/restore/restore_manager.go:405]:
```suggestion
	} else if tc.Spec.TiKV.Replicas != metaInfo.KubernetesMeta.TiDBCluster.Spec.TiKV.Replicas {
```

File: pkg/fedvolumebackup/restore/restore_manager.go [hunk 3]
```diff
@@ -290,15 +297,27 @@ func (rm *restoreManager) waitRestoreTiKVComplete(volumeRestore *v1alpha1.Volume
...
+			continue
+		}
 		if !pingcapv1alpha1.IsRestoreTiKVComplete(restoreMember.restore) {
 			return controller.IgnoreErrorf("restore member %s of cluster %s is not tikv complete", restoreMemberName, k8sClusterName)
 		}
 	}
-	// restore tikv complete
-	if !v1alpha1.IsVolumeRestoreTiKVComplete(volumeRestore) {
+
+	klog.InfoS("all warm up tasks are completed.", "restore", volumeRestore.Name, "wal-check-only-completed", walCheckCompleted, "total", len(restoreMembers))
```
Comment (WangLe1321) [inline pkg/fedvolumebackup/restore/restore_manager.go:315]: If it's async warmup, warmup tasks may not complete here. I suggest you move the logic of check wal complete to a separate function between sync warmup status and check tikv complete.

**Review by WangLe1321 — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Pull request merged**